data-formats/src/constants/mod.rs

```diff
@@ -425,7 +425,8 @@ pub enum ErrorCode {
 #[repr(i8)]
 #[non_exhaustive]
 pub enum MfgStringType {
-    SerialNumber = 0,
+    SerialNumber,
```

Comment:
This is a different enum from the one at `util/.../manufacturing_server.rs` and it does need the constants.

Reply:
Can we centralise them in an include file rather than duplicating?

Reply:
No because the ones in `data-formats` are going to be sent over network and the serialization is different from the ones at `util`, which are serialized into files (for user configuration).

Reply:
> This is a different enum from the one at `util/.../manufacturing_server.rs` and it does need the constants.

ahh right! since we are reading values of these enum members directly without initializing we need constants! Learned this new that Rust wont assign values by default if not initialised. But I wonder why compiler didnt give any error?

Reply:
> > This is a different enum from the one at `util/.../manufacturing_server.rs` and it does need the constants.
> 
> ahh right! since we are reading values of these enum members directly without initializing we need constants! Learned this new that Rust wont assign values by default if not initialised. But I wonder why compiler didnt give any error?

The constants are needed because of the `#[repr(i8)]` which forces the enum to have the same data layout representation as it would have in C. If we don't set the constant there might be some undefined behaviour going on.